### DISCUSSION TITLE
Conflicting lifecycle mapping error after updating to the latest (2.0.3.20230228-2348) version

### DISCUSSION THREAD

**Discussion by grosenberg** [eclipse-m2e/m2e-core#1290] [Q&A]
Updated Eclipse (2022-12) to the most recent M2E version and now building of my plugin [IndentGuide](https://github.com/grosenberg/IndentGuide) fails.

The specific error is 

> Conflicting lifecycle mapping metadata (project packaging type="eclipse-plugin"): Mapping defined in 
> 'org.eclipse.m2e.pde.connector_2.1.2.20221218-1726 [793]' and 'org.sonatype.tycho.m2e_0.10.0.20220926-1324 [803]'.
> To enable full functionality, remove the conflicting mapping and run Maven->Update Project Configuration.
> pom.xml	/indentguide.plugin	line 10	Maven Project Build Lifecycle Mapping Problem

Removing the Sonatype Tycho connector removes the error, but cannot build due to a failure to launch problem

> Launch configuration Executing clean in D__DevFiles_Eclipse_Tools_IndentGuide_net.certiv.tools.indentguide 
> references non-existing project net.certiv.tools.indentguide.

1) How to resolve the Sonatype Tycho connector conflict?
2) Is there a best practices/migration guide (the release notes seem to be a bit cryptic in this regard)?

Thanks

![Installed Software](https://user-images.githubusercontent.com/1757970/222590169-4e6ff8f9-c4b2-4a69-b1f1-2fe8f8bba137.jpg)

**Comment by laeubi** [comment]
> How to resolve the Sonatype Tycho connector conflict?

You should not need any special connector for using Tycho

> Removing the Sonatype Tycho connector removes the error, but cannot build due to a failure to launch problem

This seems unrelated, where / when exactly do you see this error?

**Comment by grosenberg** [comment]
Right click on the pom.xml in the parent project, then `Run As -> Maven clean`

Effectively the same message for `Maven build` and `Maven verify`.  Same if a named launch configuration is created with the goals of `clean verify` 

To be certain, no other launch configurations reference this plugin project.

Thanks.

**Reply by laeubi** [reply]
The error says it references a non existing project (net.certiv.tools.indentguide) so maybe you can clean up your launch configs and start with a fresh one?

**Reply by grosenberg** [reply]
Yes, did that previously to no effect. 

If I rollback to the prior M2E plugin version (and have the Sonotype connector installed), the launch works as intended/all bundles are found, built, and assembled correctly.

BTW, just noticed that with the new M2E version, getting an info mark against the 	`<packaging>eclipse-repository</packaging>` line within the repository site bundle.

The info message
> Plugin execution not covered by lifecycle configuration: org.apache.maven.plugins:maven-clean-plugin:2.5:clean 
> (execution: default-clean-1, phase: initialize)

**Reply by laeubi** [reply]
Well the error you are seeing is completely unrelated to life-cycle mappings, as this starts an (external) maven build, so it needs to be something else ... have you enabled any special options or just can you try with a fresh download of eclipse with no extra plugins installed?

**Reply by grosenberg** [reply]
> have you enabled any special options 

None that I know of -- very  plain Eclipse/Maven/Tycho user.

> can you try with a fresh download of eclipse with no extra plugins installed?

Uninstalling all extras in the current installation had no effect. Will try a complete reinstall next. 

Thanks.

**Comment by laeubi** [comment]
@grosenberg I can verify that this happens under windows also for me without any special configuration, I created:

- https://github.com/eclipse-m2e/m2e-core/issues/1293

**Reply by laeubi** [reply]
By the way the current workaround is to enable the java nature for the project of the pom.xml, nerveless we are working on a solution!

**Reply by HannesWell** [reply]
This should now be fixed with https://github.com/eclipse-m2e/m2e-core/pull/1294.
which will be available in about 20min at:
https://download.eclipse.org/technology/m2e/snapshots/latest/

**Comment by grosenberg** [comment]
Can confirm that the build now completes correctly.

However, a possibly related info marker remains -- not sure if this existed earlier or not

Occurs in the  repository plugin  POM

![EclipseM2eInfoError](https://user-images.githubusercontent.com/1757970/224465043-52a75668-2aa8-4fc1-b96e-5afadba337f4.jpg)

The marker tooltip reads

> Plugin execution not covered by lifecycle configuration: org.apache.maven.plugins:maven-clean-plugin:2.5:clean (execution: default-clean-1, phase: initialize)

Notably, the prompted fix (hover over `<packaging>`) leads to recommending the installation of the Sonatype tycho bridge connector!

Thanks.
